File: JobsProject/lib/models/job.dart
class Job {
  final String id;
  final String title;
  final String description;
  final String requirements;
  final String companyName;
  final String companyLogo;
  final String location;
  final String salary;
  final String ageRange;
  final String genderRequirement;
  final int weight;
  final String countryId;
  final String? whatsappPhone;

  Job({
    required this.id,
    required this.title,
    required this.description,
    required this.requirements,
    required this.companyName,
    required this.companyLogo,
    required this.location,
    required this.salary,
    required this.ageRange,
    required this.genderRequirement,
    required this.weight,
    required this.countryId,
    this.whatsappPhone,
  });

  factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'] as String,
      title: json['title'] as String,
      description: json['description'] as String,
      requirements: json['requirements'] as String,
      companyName: json['companyName'] as String,
      companyLogo: json['companyLogo'] as String,
      location: json['location'] as String,
      salary: json['salary'] as String,
      ageRange: json['ageRange'] as String,
      genderRequirement: json['genderRequirement'] as String,
      weight: json['weight'] as int,
      countryId: json['countryId'] as String,
      whatsappPhone: json['whatsappPhone'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'requirements': requirements,
      'companyName': companyName,
      'companyLogo': companyLogo,
      'location': location,
      'salary': salary,
      'ageRange': ageRange,
      'genderRequirement': genderRequirement,
      'weight': weight,
      'countryId': countryId,
      'whatsappPhone': whatsappPhone,
    };
  }
}

File: JobsProject-Backend/src/models/Job.js
const pool = require('../config/database');

class Job {
  static async getAll(filters = {}) {
    let query = 'SELECT * FROM jobs WHERE 1=1';
    const params = [];

    if (filters.country) {
      query += ' AND country_id = ?';
      params.push(filters.country);
    }

    if (filters.ageMin !== undefined && filters.ageMax !== undefined) {
      query += ' AND age_range LIKE ?';
      params.push(`${filters.ageMin}-%`);
    }

    if (filters.gender && filters.gender !== 'all') {
      query += ' AND gender_requirement = ?';
      params.push(filters.gender);
    }

    query += ' ORDER BY weight DESC LIMIT ? OFFSET ?';
    params.push(filters.limit || 20, filters.offset || 0);

    const [rows] = await pool.query(query, params);
    return rows;
  }

  static async getById(id) {
    const [rows] = await pool.query('SELECT * FROM jobs WHERE id = ?', [id]);
    return rows[0];
  }

  static async create(jobData) {
    const {
      title,
      description,
      requirements,
      companyName,
      companyLogo,
      location,
      salary,
      ageRange,
      genderRequirement,
      weight,
      countryId,
      whatsappPhone,
    } = jobData;

    const [result] = await pool.query(
      `INSERT INTO jobs
       (title, description, requirements, company_name, company_logo, location, salary, age_range, gender_requirement, weight, country_id, whatsapp_phone)
       VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
      [title, description, requirements, companyName, companyLogo, location, salary, ageRange, genderRequirement, weight, countryId, whatsappPhone]
    );

    return result.insertId;
  }

  static async update(id, jobData) {
    const fields = [];
    const values = [];

    Object.entries(jobData).forEach(([key, value]) => {
      fields.push(`${key} = ?`);
      values.push(value);
    });

    values.push(id);

    const [result] = await pool.query(
      `UPDATE jobs SET ${fields.join(', ')} WHERE id = ?`,
      values
    );

    return result.affectedRows > 0;
  }

  static async delete(id) {
    const [result] = await pool.query('DELETE FROM jobs WHERE id = ?', [id]);
    return result.affectedRows > 0;
  }
}

module.exports = Job;
